jsf-showcase-webapp/src/main/java/com/liferay/faces/showcase/converter/CountryConverter.java

```diff
@@ -65,14 +65,20 @@ public String getAsString(FacesContext facesContext, UIComponent uiComponent, Ob
 
 	protected Map<Long, Country> getCountryMap(FacesContext facesContext) {
 
+		// First check without locking (not yet thread-safe)
 		if (countryMap == null) {
 
 			synchronized (this) {
```

Comment:
@stiemannkj1 I am going to assume that since the signature for Converter has methods that take `FacesContext` that it is a singleton that is effectively stored in application scope. If that is the case, then `synchronized(this)` is sufficient. If it is not a case, then it is not sufficient. Just to make sure, please put a `System.err.println` in the converter's constructor to see when (and possibly how often) it gets created.